aderyn_core/src/detect/util/identifiers_that_reference_state_variables.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::{
+    ast::NodeID,
+    capture,
+    context::workspace_context::{ASTNode, WorkspaceContext},
+    detect::detector::{Detector, IssueSeverity},
+};
+
+#[derive(Default)]
+pub struct IdentifiersThatReferenceStateVariableDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize), NodeID>,
+}
+
+impl Detector for IdentifiersThatReferenceStateVariableDetector {
+    fn detect(
+        &mut self,
+        context: &WorkspaceContext,
+        _: &[NodeID],
+        using: &[NodeID],
+    ) -> std::prelude::v1::Result<bool, Box<dyn Error>> {
```

Comment:
Please use just `Result`

Reply:
If we have a `bot_example` that uses this Detector it would be nice : )

Reply:
Agreed.